graphql_api/tests/test_test_analytics.py

```diff
@@ -0,0 +1,827 @@
+import datetime
+
+from django.test import TransactionTestCase
+from shared.django_apps.reports.tests.factories import FlakeFactory
+
+from codecov_auth.tests.factories import OwnerFactory
+from core.tests.factories import (
+    RepositoryFactory,
+)
+from reports.tests.factories import (
+    DailyTestRollupFactory,
+    RepositoryFlagFactory,
+    TestFactory,
+    TestFlagBridgeFactory,
+)
+
+from .helper import GraphQLTestHelper
+
+
+class TestAnalyticsTestCase(GraphQLTestHelper, TransactionTestCase):
+    def setUp(self) -> None:
+        self.owner = OwnerFactory(username="codecov-user")
+        self.repo = RepositoryFactory(
+            author=self.owner, name="testRepoName", active=True
+        )
+
+    query_builder = """
+    query TestAnalytics($name: String!){
+        me {
+            owner {
+                repository(name: $name) {
+                    __typename
+                    ... on Repository {
+                        testAnalytics {
+                            %s
+                        }
+                    }
+                    ... on ResolverError {
+                        message
+                    }
+                }
+            }
+        }
+    }
+    """
+
+    def fetch_test_analytics(self, name, fields=None):
+        data = self.gql_request(
+            self.query_builder % fields,
+            owner=self.owner,
+            variables={"name": name},
+        )
+        return data["me"]["owner"]["repository"]["testAnalytics"]
+
+    def test_repository_test_analytics_typename(self):
+        response = self.gql_request(
+            """
+            query($owner: String!, $repo: String!) {
+              owner(username: $owner) {
+                repository(name: $repo) {
+                  ... on Repository {
+                    testAnalytics {
+                      __typename
+                    }
+                  }
+                  ... on ResolverError {
+                    message
+                  }
+                }
+              }
+            }
+            """,
+            owner=self.owner,
+            variables={"owner": self.owner.username, "repo": self.repo.name},
+        )
+
+        assert (
+            response["owner"]["repository"]["testAnalytics"]["__typename"]
+            == "TestAnalytics"
+        )
+
+    def test_test_results(self) -> None:
```

Comment:
everything from this test onwards is copy/pasted from existing `repository_test.py` tests related to the existing `repository.testResults`, `repository.testResultsAggregates` and `repository.flakeAggregates` fields, but now nested one level deep under `repository.testAnalytics`.

The duplicated tests in `repository_test.py` will get removed after the relevant frontend pieces get cut over